lib/core/storage.ts

```diff
@@ -40,14 +53,38 @@ export class WithTimeout implements IStorage {
     for await (const res of this.storage.search("*", undefined, undefined, limit)) {
       yield res
         .map(timeout)
-        .filter((promise) => regex.test(promise.id))
+        .filter(isDurablePromise)
         .filter((promise) => states.includes(promise.state))
+        .filter((promise) => regex.test(promise.id))
         .filter((promise) => tagEntries.every(([k, v]) => promise.tags?.[k] == v));
     }
   }
 }
 
-function timeout<P extends DurablePromise | undefined>(promise: P): P | TimedoutPromise {
+export class WithTimeoutSchedules implements IScheduleStorage {
```

Comment:
I don't think we need a WithTimeout for schedule. We wrap the local promise one, but for schedules we should be able to just use the local one.